python_modules/dagster/dagster/core/execution/context/invocation.py

```diff
@@ -231,8 +239,15 @@ def bind(
             alias=solid_def_or_invocation.given_alias
             if isinstance(solid_def_or_invocation, PendingNodeInvocation)
             else None,
+            user_events=self._user_events,
         )
 
+    def get_events(self) -> List[UserEvent]:
+        return self._user_events
+
+    def scrub_events(self) -> None:
+        self._user_events = []
+
```

Comment:
Are we okay with this existing? The idea is to serve use cases where someone is using the same context in multiple ops, and to provide them with a way of clearing the cache between invocations

Reply:
My instinct would be to keep this out. A world where contexts are mutable and intended for reuse across multiple ops feels like it could open itself up to all kinds of unanticipated problems.